test/functional/ui/cmdline_spec.lua

```diff
@@ -1485,31 +1485,28 @@ describe('cmdheight=0', function()
   it('when substitute text', function()
     command('set cmdheight=0 noruler laststatus=3')
     feed('ifoo<ESC>')
-    screen:expect {
-      grid = [[
+    screen:try_resize(screen._width, 6)
+    screen:expect([[
       fo^o                      |
-      {1:~                        }|*3
+      {1:~                        }|*4
       {3:[No Name] [+]            }|
-    ]],
-    }
+    ]])
 
     feed(':%s/foo/bar/gc<CR>')
-    screen:expect {
-      grid = [[
+    screen:expect([[
       {2:foo}                      |
-      {1:~                        }|*3
-      {6:replace wi...q/l/^E/^Y)?}^ |
-    ]],
-    }
+      {3:                         }|
+                               |*2
```

Comment:
This is a `:set cmdheight=0` issue that happens somewhere in the incsearch path. Whenever cmdheight == 0 and we type `/foo`, `?foo` or `:%s/foo` where foo exceeds the number of `Columns` we get a `cmdline/msg_row` conflict, resulting in the cleared row and extra scrolled line here. I think it's related to:
https://github.com/neovim/neovim/blob/adcd9360dfefc7b1e1edb0e86df460e074991c8d/src/nvim/message.c#L237-L240

I could attempt to fix this but that work will be in vain assuming we remove the message grid after #27855.

Reply:
I probably had to reproduce and report the issue with multi-line commands and broken refresh (when switching between command line history e.g. by `<C-Up>`) a few days/weeks ago, but it wasn't necessarily related to this request. I am still not sure if you're talking about the same thing.